flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1103,23 +1093,56 @@ private void archiveExecutionGraph(ExecutionGraphInfo executionGraphInfo) {
                     executionGraphInfo.getArchivedExecutionGraph().getJobID(),
                     e);
         }
+    }
+
+    private CompletableFuture<Acknowledge> archiveExecutionGraph(
+            ExecutionGraphInfo executionGraphInfo) {
 
         // do not create an archive for suspended jobs, as this would eventually lead to multiple
         // archive attempts which we currently do not support
```

Comment:
This comment is obsolete here. The method itself does not require this invariant. It's more sufficient at the code location where this method is called.